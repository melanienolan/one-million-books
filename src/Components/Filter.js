import React from 'react';
import PropTypes from 'prop-types';

const Filter = props => {
  const { filterType, filterCategories, activeFilter } = props;
  const renderList = () => {
    return filterCategories.map((category, i) => {
      return (
        <option key={i} value={category}>
          {category}
        </option>
      );
    });
  };
  return (
    <div>
      <form className="form" action="">
        <label className="form--label">
          <span className="form--text">{`Filter by ${filterType}`}</span>
          <select
            className="form--input"
            onChange={e => props.updateFilters(e, filterType)}
            value={props.activeFilter}
            name="filter"
            id="">
            {renderList()}
          </select>
        </label>
      </form>
    </div>
  );
};

Filter.propTypes = {
  filterCategories: PropTypes.array,
  filterType: PropTypes.string,
  activeFilter: PropTypes.string,
  updateFilters: PropTypes.func
};

export default Filter;
