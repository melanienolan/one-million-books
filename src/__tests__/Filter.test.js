import React from 'react';
import Filter from '../Components/Filter';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import db from '../mockDB';

Enzyme.configure({ adapter: new Adapter() });
const { genres, specialDates } = db;

it('renders filter component without crashing', () => {
  // const {genres} = db;
  const filter = genres[4];
  Enzyme.shallow(
    <Filter filterCategories={genres} filterType="genre" activeFilter={filter} />
  );
});
it('renders genre filter component with All', () => {
  const filter = genres[0];
  const filterComponent = Enzyme.shallow(
    <Filter filterCategories={genres} filterType="genre" activeFilter={filter} />
  );
  expect(toJson(filterComponent)).toMatchSnapshot();
});
it('renders special dates filter component with Halloween', () => {
  const filter = specialDates[1];
  const filterComponent = Enzyme.shallow(
    <Filter
      filterCategories={specialDates}
      filterType="specialDates"
      activeFilter={filter}
    />
  );
  expect(toJson(filterComponent)).toMatchSnapshot();
});
