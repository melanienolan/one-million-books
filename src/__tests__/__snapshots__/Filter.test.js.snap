// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders genre filter component with All 1`] = `
<div>
  <form
    action=""
    className="form"
  >
    <label
      className="form--label"
    >
      <span
        className="form--text"
      >
        Genre
      </span>
      <select
        className="form--input"
        id=""
        name="filter"
        onChange={[Function]}
        value="All"
      >
        <option
          key="0"
          value="All"
        >
          All
        </option>
        <option
          key="1"
          value="Finance"
        >
          Finance
        </option>
        <option
          key="2"
          value="Science fiction"
        >
          Science fiction
        </option>
        <option
          key="3"
          value="Satire"
        >
          Satire
        </option>
        <option
          key="4"
          value="Drama"
        >
          Drama
        </option>
        <option
          key="5"
          value="Action and Adventure"
        >
          Action and Adventure
        </option>
        <option
          key="6"
          value="Romance"
        >
          Romance
        </option>
        <option
          key="7"
          value="Mystery"
        >
          Mystery
        </option>
        <option
          key="8"
          value="Horror"
        >
          Horror
        </option>
        <option
          key="9"
          value="Self help"
        >
          Self help
        </option>
        <option
          key="10"
          value="Health"
        >
          Health
        </option>
        <option
          key="11"
          value="Guide"
        >
          Guide
        </option>
        <option
          key="12"
          value="Travel"
        >
          Travel
        </option>
        <option
          key="13"
          value="Children"
        >
          Children
        </option>
        <option
          key="14"
          value="Religion, Spirituality & New Age"
        >
          Religion, Spirituality & New Age
        </option>
        <option
          key="15"
          value="Science"
        >
          Science
        </option>
        <option
          key="16"
          value="History"
        >
          History
        </option>
        <option
          key="17"
          value="Math"
        >
          Math
        </option>
        <option
          key="18"
          value="Anthology"
        >
          Anthology
        </option>
        <option
          key="19"
          value="Poetry"
        >
          Poetry
        </option>
        <option
          key="20"
          value="Encyclopedias"
        >
          Encyclopedias
        </option>
        <option
          key="21"
          value="Dictionaries"
        >
          Dictionaries
        </option>
        <option
          key="22"
          value="Comics"
        >
          Comics
        </option>
        <option
          key="23"
          value="Art"
        >
          Art
        </option>
        <option
          key="24"
          value="Cookbooks"
        >
          Cookbooks
        </option>
        <option
          key="25"
          value="Diaries"
        >
          Diaries
        </option>
        <option
          key="26"
          value="Journals"
        >
          Journals
        </option>
        <option
          key="27"
          value="Prayer books"
        >
          Prayer books
        </option>
        <option
          key="28"
          value="Series"
        >
          Series
        </option>
        <option
          key="29"
          value="Trilogy"
        >
          Trilogy
        </option>
        <option
          key="30"
          value="Biographies"
        >
          Biographies
        </option>
        <option
          key="31"
          value="Autobiographies"
        >
          Autobiographies
        </option>
        <option
          key="32"
          value="Fantasy"
        >
          Fantasy
        </option>
      </select>
    </label>
  </form>
</div>
`;

exports[`renders special dates filter component with Halloween 1`] = `
<div>
  <form
    action=""
    className="form"
  >
    <label
      className="form--label"
    >
      <span
        className="form--text"
      >
        Special Dates
      </span>
      <select
        className="form--input"
        id=""
        name="filter"
        onChange={[Function]}
        value="Halloween"
      >
        <option
          key="0"
          value="All"
        >
          All
        </option>
        <option
          key="1"
          value="Halloween"
        >
          Halloween
        </option>
        <option
          key="2"
          value="Last Friday of Month"
        >
          Last Friday of Month
        </option>
        <option
          key="3"
          value="Halloween and Last Friday"
        >
          Halloween and Last Friday
        </option>
      </select>
    </label>
  </form>
</div>
`;
